packages/material-ui-lab/src/Rating/Rating.js

```diff
@@ -0,0 +1,475 @@
+
+    if (onChangeActive && focus !== newFocus) {
+      onChangeActive(event, newFocus);
+    }
+  };
+
+  const item = (propsItem, state) => {
+    const id = `${name}-${String(propsItem.value).replace('.', '-')}`;
+    const container = (
+      <IconContainerComponent
+        {...propsItem}
+        className={clsx(classes.icon, {
+          [classes.iconEmpty]: !state.filled,
+          [classes.iconFilled]: state.filled,
+          [classes.iconHover]: state.hover,
+          [classes.iconFocus]: state.focus,
+          [classes.iconActive]: state.active,
+        })}
+      >
+        {emptyIcon && !state.filled ? emptyIcon : icon}
+      </IconContainerComponent>
+    );
+
+    if (readOnly || disabled) {
+      return <React.Fragment key={propsItem.value}>{container}</React.Fragment>;
+    }
+
+    return (
+      <React.Fragment key={propsItem.value}>
+        <label className={classes.label} htmlFor={id}>
+          {container}
+          <span className={classes.visuallyhidden}>{getLabelText(propsItem.value)}</span>
+        </label>
+        <input
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onChange={handleChange}
+          value={propsItem.value}
+          id={id}
+          type="radio"
+          name={name}
+          checked={state.checked}
+          className={classes.visuallyhidden}
+        />
+      </React.Fragment>
+    );
+  };
+
+  return (
+    <Component
+      ref={handleRef}
+      onMouseMove={handleMouseMove}
+      onMouseLeave={handleMouseLeave}
+      className={clsx(
+        classes.root,
+        {
+          [classes[`size${capitalize(size)}`]]: size !== 'medium',
+          [classes.disabled]: disabled,
+          [classes.focusVisible]: focusVisible,
+          [classes.readOnly]: readOnly,
+        },
+        className,
+      )}
+      {...other}
+    >
+      {!readOnly && !disabled && value == null && (
+        <React.Fragment>
+          <input
+            value="0"
+            id={`${name}-0`}
+            type="radio"
+            name={name}
+            defaultChecked
+            className={classes.visuallyhidden}
+          />
+          <label htmlFor={`${name}-0`} className={classes.pristine}>
+            <span className={classes.visuallyhidden}>{getLabelText(0)}</span>
+          </label>
+        </React.Fragment>
+      )}
+      {Array.from(new Array(max)).map((_, index) => {
+        const itemValue = index + 1;
+
+        if (precision < 1) {
+          const items = Array.from(new Array(1 / precision));
+          return (
+            <div
+              key={itemValue}
+              className={clsx(classes.decimal, {
+                [classes.iconActive]:
+                  itemValue === Math.ceil(value) && (hover !== -1 || focus !== -1),
+              })}
+            >
+              {items.map(($, indexDecimal) => {
+                const itemDeciamlValue = itemValue - 1 + (indexDecimal + 1) * precision;
+
+                return item(
```

Comment:
Why render functions over reacts usual composition model?

Reply:
I think that the fewer components we have to create the better. In this case, I couldn't find a value in creating one. Alternatively, we could consider if it's worth moving outside of the main Rating function scope.

Reply:
> I think that the fewer components we have to create the better.

What makes you think that?

Reply:
because the alternative is functions, I think that it's simpler.